src/Workspaces/Core/Portable/Workspace/Solution/CompilationOutputFilePaths.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis
+{
+    /// <summary>
+    /// Paths of files produced by the compilation.
+    /// </summary>
+    public readonly struct CompilationOutputFilePaths : IEquatable<CompilationOutputFilePaths>, IObjectWritable
```

Comment:
is a struct a potentil concern in the future.  i know we've tended to avoid them in these APIs because chaging them over time can be harder.

Reply:
Why is it harder to add a field to a struct?